### DISCUSSION TITLE
Filtering delay

### DISCUSSION THREAD

**Discussion by AnejVovcak** [dj-fiorex/angular2-smart-table#141] [Q&A]
Hey everyone!

I'm using your library with the ServerDataSource class, and it's been great for enabling date field filtering by users. However, I've run into a bit of an issue where the text in the input field disappears while users are typing. Oddly, I noticed this behavior in your official documentation as well.

I'm curious if there's a way to introduce a delay in the GET request to help with this problem. Any suggestions would be awesome!

Thanks for your quick response and for maintaining this awesome library!

**Comment by uap-universe** [comment]
Unfortunately we mostly inherited the ServerDataSource and only tried to stabilize it somehow without diving too deep into it. 

I am quite sure that we would somehow need to debounce this call to fix the issue:

https://github.com/dj-fiorex/angular2-smart-table/blob/c07e6b363935a080b83db1b96158de5be4dbb630/projects/angular2-smart-table/src/lib/lib/data-source/server/server.data-source.ts#L93

It would require some more analysis.

**Reply by AnejVovcak** [reply]
Thank you for your response! Yes, I agree. Let me know if you are planning to look into this.

**Comment by uap-universe** [comment]
I have created ticket #144 for it, we can continue discussion there.
